[EDU-6046] Create Azion Lib docs in English (#1479)
aziontech/docs: src/content/docs/en/pages/devtools/azion-lib/usage/sql.mdx

|----------|-------------|
| `AZION_TOKEN` | Your Azion API token. |
| `AZION_DEBUG` | Enable debug mode (true/false). |

:::tip
Setting `AZION_DEBUG` to `true` enables **Debug mode**. This mode provides detailed logging of the API requests and responses.
:::

If you want to create a specific `client` for interacting with **SQL**, create one by calling the `createClient` method from the library:

```typescript
import { createClient } from 'azion/purge';
import type { AzionSQLClient, AzionSQLResponse, AzionSQL } from 'azion/purge';

const client: AzionSQLClient = createClient({ token: 'your-api-token', options: { debug: true } });

const { data, error } = await client.createDatabase('my-new-db');
if (data) {
  console.log(`Database created with ID: ${data.id}`);
} else {
  console.error('Failed to create database', error);
}
```

The `createClient` method has the following **parameters** and **return value**:

**Parameters:**

| Parameter | Type | Description |
|-----------|------|-------------|
| `config` | `Partial<{ token: string; options?: OptionsParams }>` | Configuration options for the SQL client. |

**Returns:**

| Return Type | Description |
|-------------|-------------|

> | [`AzionSQLClient`](#azionSQLclient) | An object with methods to interact with SQL. |

Reviewer: Aqui SQL é o client, né? Faz sentido o original

:::note
In the following examples, the methods are called directly without the creation of a `client`. For more information on how to interact with services and products using a `client`, check the [Azion Lib Client documentation](/en/documentation/products/azion-lib/client/).
:::

> 
> ## Usage

Reviewer: ```suggestion

---

## Usage
```

### `createDatabase`

Creates a new database.

**Parameters:**

> **Parameters:**
> 

Reviewer: ```suggestion
```

Reviewer: duplicate

| `statements` | `string[]` | Array of SQL statements to execute. |
| `options` | `AzionClientOptions` (optional) | Optional parameters for the query. |

**Returns:**

| Return Type | Description |
|-------------|-------------|
| `Promise<AzionDatabaseResponse<AzionDatabaseQueryResponse>>` | Query result object or error. |

**Example:**

```typescript
import { useQuery, AzionDatabaseQueryResponse, AzionDatabaseResponse } from 'azion/sql';

const { data: result, error }: AzionDatabaseResponse<AzionDatabaseQueryResponse> = await useQuery(
  'my-db',
  ['SELECT * FROM users'],
  {
    debug: true,
  },
);
if (result) {
  console.log(`Query executed. Rows returned: ${result.rows.length}`);
} else {
  console.error('Query execution failed', error);
}
```

### `useExecute`

Executes a set of SQL statements on a specific database.

**Parameters:**

**Parameters:**

| Parameter | Type | Description |
|-----------|------|-------------|
| `name` | `string` | Name of the database to execute the statements on. |
| `statements` | `string[]` | Array of SQL statements to execute. |
| `options` | `AzionClientOptions` (optional) | Optional parameters for the execution. |

> **Returns:**

Reviewer 2: ```suggestion
**Returns**:
```

| Return Type | Description |
|-------------|-------------|
| `Promise<AzionDatabaseResponse<AzionDatabaseQueryResponse>>` | Execution result object or error. |

> **Example:**

Reviewer 2: ```suggestion
**Example**:
```

```typescript
import { useExecute, AzionDatabaseQueryResponse } from 'azion/sql';

const result: AzionDatabaseQueryResponse | null = await useExecute(
  'my-db',
  ['INSERT INTO users (name) VALUES ("John")'],
  {
    debug: true,
  },
);
if (result?.state === 'executed') {
  console.log('Executed with success');
} else {
  console.error('Execution failed');
}
```

> 
> ## Types

Reviewer: ```suggestion

---

## Types
```

> These are the types used by the **SQL Library** and its methods:

Reviewer 2: ```suggestion
These are the types used by the **SQL** library and its methods:
```

| Parameter | Type | Description |
|-----------|------|-------------|
| `token` | `string` (optional) | Your Azion API token. |
| `options` | `AzionClientOptions` (optional) | Optional parameters for the client configuration. |

### `AzionSQLClient`

An object with methods to interact with SQL databases.

| Method | Parameters | Return Type |
|--------|------------|-------------|
| `createDatabase` | `name: string` | `Promise<AzionDatabaseResponse<AzionDatabase>>` |
| `deleteDatabase` | `id: number` | `Promise<AzionDatabaseResponse<AzionDatabaseDeleteResponse>>` |
| `getDatabase` | `name: string` | `Promise<AzionDatabaseResponse<AzionDatabase>>` |
| `getDatabases` | `params?: AzionDatabaseCollectionOptions` | `Promise<AzionDatabaseResponse<AzionDatabaseCollections>>` |
| `useQuery` | `name: string, statements: string[], options?: AzionClientOptions` | `Promise<AzionDatabaseResponse<AzionDatabaseQueryResponse>>` |
| `useExecute` | `name: string, statements: string[], options?: AzionClientOptions` | `Promise<AzionDatabaseResponse<AzionDatabaseExecutionResponse>>` |

### `AzionDatabase`

The database object.

| Property | Type | Description |
|----------|------|-------------|

> | `id` | `number` | Unique identifier for the database. |

Reviewer 2: ```suggestion
| `id` | `number` | A unique identifier for the database. |
```

| `name` | `string` | The name of the database. |
| `clientId` | `string` | The client ID associated with the database. |
| `status` | `string` | The current status of the database. |

> | `createdAt` | `string` | Timestamp of when the database was created. |

Reviewer 2: ```suggestion
| `createdAt` | `string` | The timestamp of when the database was created. |
```

> | `updatedAt` | `string` | Timestamp of when the database was last updated. |

Reviewer 2: ```suggestion
| `updatedAt` | `string` | The timestamp of when the database was last updated. |
```

> | `deletedAt` | `string | null` | Timestamp of when the database was deleted, or null if not deleted. |

Reviewer 2: ```suggestion
| `deletedAt` | `string | null` | The timestamp of when the database was deleted, or null if not deleted. |
```

> | `deletedAt` | `string | null` | Timestamp of when the database was deleted, or null if not deleted. |

Reviewer: Está certo o | `string | null` | 
Parece estar adicionando uma coluna adicional, validar se não quebra a tabela.

| `query` | `(statements: string[], options?: AzionClientOptions) => Promise<AzionDatabaseResponse<AzionDatabaseQueryResponse>>` | Executes a query on the database. |
| `execute` | `(statements: string[], options?: AzionClientOptions) => Promise<AzionDatabaseResponse<AzionDatabaseExecutionResponse>>` | Executes a SQL statement on the database. |
| `getTables` | `(options?: AzionClientOptions) => Promise<AzionDatabaseResponse<AzionDatabaseQueryResponse>>` | Retrieves a list of tables in the database. |

### `AzionDatabaseResponse<T>`

The response object from a database operation.

| Property | Type | Description |
|----------|------|-------------|
| `data` | `T` (optional) | The data returned from the operation. |
| `error` | `{ message: string, operation: string }` (optional) | The error details if the operation fails. |

### `QueryResult`

| Property | Type | Description |
|----------|------|-------------|
| `state` | `string` | The state of the query execution. |
| `columns` | `string[]` | An array of column names. |
| `statement` | `string` | The SQL statement executed. |

> | `rows` | `(number | string)[][]` | A 2D array of rows, where each row is an array of values. |

Reviewer: validar que não está quebrando a tabela.

### `AzionClientOptions`

Optional parameters for the client configuration.

| Property | Type | Description |
|----------|------|-------------|
| `debug` | `boolean` (optional) | Enables debug mode. |

> | `force` | `boolean` (optional) | Forces the operation to execute. |

Reviewer 2: ```suggestion
| `force` | `boolean` (optional) | Forces the operation to be executed. |
```

### `AzionDatabaseCollectionOptions`

Optional parameters for filtering and pagination.

| Property | Type | Description |
|----------|------|-------------|
| `ordering` | `string` (optional) | Specifies the ordering of the results. |
| `page` | `number` (optional) | The page number for pagination. |
| `page_size` | `number` (optional) | The number of items per page. |
| `search` | `string` (optional) | A search query to filter the results. |

### `AzionDatabaseQueryResponse`

The response object from a query execution.

| `state` | `'executed' | 'pending' | 'executed-runtime' | 'failed'` | The state of the query execution. |
| `data` | `QueryResult[]` | The data returned from the operation. |
| `toObject` | `() => JsonObjectQueryExecutionResponse` (optional) | A method to convert the response to a JSON object. |

### `AzionDatabaseExecutionResponse`

The response object from a query execution.

| Property | Type | Description |
|----------|------|-------------|

> | `state` | `'executed' | 'pending' | 'executed-runtime' | 'failed'` | The state of the query execution. |

Reviewer: validar, pode estar quebrando a tabela

| `data` | `QueryResult[]` | The data returned from the operation. |
| `toObject` | `() => JsonObjectQueryExecutionResponse` (optional) | A method to convert the response to a JSON object. |

### `AzionQueryExecutionParams`

Parameters for query execution.

| Property | Type | Description |
|----------|------|-------------|
| `statements` | `string[]` | An array of SQL statements. |

> | `params` | `(AzionQueryParams | `Record<string, AzionQueryParams>)[]` (optional) | An array of query parameters. |

Reviewer: validar, pode estar quebrando a tabela.
